### ISSUE TITLE
OpenCL on Mac produces incorrect training results

### ISSUE THREAD

**Issue by monogenea** [plaidml/plaidml#193]
Hi guys. After following the instructions for installing plaidml on my iMac Pro, I originally chose `opencl_amd_amd_radeon_vega_56`, confirmed the selection in the ~/.plaidml file, and successfully deployed plaidml as the backend for Keras using both R and Python. Runtime of CNN benchmarks was substantially shorter, as expected.

The problem, however, was that the actual optimisation was going mental, with training loss increasing, rather than decreasing, indefinitely and concluding with reported missing accuracy - overall a strange behaviour, as you see.

After looking into many different potential causes, I found that replacing the `opencl_amd_amd_radeon_vega_56` option by the `metal_amd_radeon_vega_56` one maintained the performance gain on GPU usage and also rendered accurate loss minimisation, with overall accuracy comparable to benchmarks (i.e. Keras MNIST, CIFAR10 and the like).

Maybe there was something obvious about the options that I missed; it would be great to hear from other users with the same specs.

**Comment by Cloudcatcher888** [comment]
I met the same problem. My loss went to negative infinite

**Comment by brianretford** [comment]
We are going to move metal to be the default option in the next release. Pre-mojave we were noticing significant issues in metal, but post-mojave everything seems to work much more reliably than OpenCL (which is deprecated in macOS)

**Comment by brianretford** [comment]
Next release has been updated to prefer metal over opencl for macos
